migrator/src/main/java/com/google/phonenumbers/migrator/MigrationJob.java

```diff
@@ -0,0 +1,160 @@
+   * corresponding BCP-47 region code (e.g. GB).
+   */
+  public static MigrationJob from(String numberInput, String regionCodeInput) throws IOException {
+    PhoneRegion regionCode = PhoneRegion.of(regionCodeInput);
+    RangeTree numberRanges = RangeTree.from(sanitizeNumberString(numberInput));
+    CsvTable<RangeKey> recipes = importRecipes(Paths.get(DEFAULT_RECIPES_PATH));
+
+    return new MigrationJob(numberRanges, regionCode, recipes);
+  }
+
+  /**
+   * Returns a MigrationJob instance for a given single E.164 number input, corresponding BCP-47
+   * region code (e.g. GB), and custom user recipes.csv file.
+   */
+  public static MigrationJob from(String numberInput, String regionCodeInput, String customRecipesPath)
+      throws IOException {
+    PhoneRegion regionCode = PhoneRegion.of(regionCodeInput);
+    RangeTree numberRanges = RangeTree.from(sanitizeNumberString(numberInput));
+    CsvTable<RangeKey> recipes = importRecipes(Paths.get(customRecipesPath));
+
+    return new MigrationJob(numberRanges, regionCode, recipes);
+  }
+
+  /**
+   * Returns a MigrationJob instance for a given file path containing comma separated E.164 numbers
+   * (e.g. +4477..., +4478...) along with the corresponding BCP-47 region code (e.g. GB) that
+   * numbers in the file belong to. All numbers in the file should belong to the same region.
+   */
+  public static MigrationJob from(Path fileInput, String regionCodeInput) throws IOException {
+    Scanner scanner = new Scanner(new FileReader(fileInput.toString()));
+    List<String> numbers = new ArrayList<>();
+    while (scanner.hasNext()) {
+      numbers.addAll(Arrays.asList(scanner.nextLine().split(",")));
+    }
+
+    PhoneRegion regionCode = PhoneRegion.of(regionCodeInput);
+    RangeTree numberRanges = RangeTree.from(numbers.stream().map(MigrationJob::sanitizeNumberString));
+    CsvTable<RangeKey> recipes = importRecipes(Paths.get(DEFAULT_RECIPES_PATH));
+
+    return new MigrationJob(numberRanges, regionCode, recipes);
+  }
+
+  /**
+   * Returns a MigrationJob instance for a given file path containing comma separated E.164 numbers,
+   * corresponding BCP-47 region code, and custom user recipes.csv file.
+   */
+  public static MigrationJob from(Path fileInput, String regionCodeInput, String customRecipesPath)
+      throws IOException {
+    Scanner scanner = new Scanner(new FileReader(fileInput.toString()));
+    List<String> numbers = new ArrayList<>();
+    while (scanner.hasNext()) {
+      numbers.addAll(Arrays.asList(scanner.nextLine().split(",")));
+    }
+
+    PhoneRegion regionCode = PhoneRegion.of(regionCodeInput);
+    RangeTree numberRanges = RangeTree.from(numbers.stream().map(MigrationJob::sanitizeNumberString));
+    CsvTable<RangeKey> recipes = importRecipes(Paths.get(customRecipesPath));
+
+    return new MigrationJob(numberRanges, regionCode, recipes);
+  }
+
+  private MigrationJob(RangeTree numberRange, PhoneRegion regionCode, CsvTable<RangeKey> recipesTable) {
```

Comment:
I think it is easier to read a class if the constructor classes are at the top. So I think this constructor should be first, followed by the from methods